src/m365/commands/status.ts

```diff
@@ -14,29 +14,39 @@ class StatusCommand extends Command {
   }
 
   public commandAction(logger: Logger, args: any, cb: (err?: any) => void): void {
-    if (auth.service.connected) {
-      if (this.debug) {
-        logger.logToStderr({
-          connectedAs: accessToken.getUserNameFromAccessToken(auth.service.accessTokens[auth.defaultResource].accessToken),
-          authType: AuthType[auth.service.authType],
-          accessTokens: JSON.stringify(auth.service.accessTokens, null, 2)
-        });
-      }
-      else {
-        logger.log({
-          connectedAs: accessToken.getUserNameFromAccessToken(auth.service.accessTokens[auth.defaultResource].accessToken)
-        });
-      }
-    }
-    else {
-      if (this.verbose) {
-        logger.logToStderr('Logged out from Microsoft 365');
-      }
-      else {
-        logger.log('Logged out');
-      }
-    }
-    cb();
+    auth
+      .ensureAccessToken(auth.defaultResource, logger, this.debug)
```

Comment:
hmmm I don't think this is the correct way to go as lets consider a very simple scenario like 
`Flow: login -> check status (should get account) -> logout -> check status (should get logged out status)`
lets check how it currently works 
![image](https://user-images.githubusercontent.com/58668583/184512132-5a38661c-1ab0-48f4-9556-821a90c01f14.png)
after this change the behavior is 
![image](https://user-images.githubusercontent.com/58668583/184512175-ed167c99-cc77-47c3-a30e-bc47a26d5d4f.png)

so as you may see after this change the `status` tries to login the user instead of returning the status that the user is logged out. For me this seems like a breaking change and for sure we should not target v5.x but rather v6 if we would like to do this kind of change 🤔. I know a lot of script sample scenarios that depend on this behavior and returning the 'logged out' status 😉

Reply:
@Adam-it  thanks for the review. I think I missed adding a condition. We will first have to check the auth Service status. If it is logged out, we straight away show the status. If it is connected, then the next step is to ensure that we are able to get a new token (based on the discussion in #3260, the service status check if not enough if the refresh token has expired.) I will go back and fix this.